src/transformStateWithClones.js

```diff
@@ -7,7 +7,23 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      currentState = { ...currentState, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete currentState[key];
+      }
+    } else if (action.type === 'clear') {
+      currentState = {};
+    }
+    result.push({ ...currentState });
```

Comment:
The `result.push({ ...currentState });` statement should be placed outside of the `if-else` block, at the end of each loop cycle. This ensures that the state is captured after all actions are processed, as per the additional prompt rules.